projects/benqi-lending/functions/getMarketBorrowLimit.spec.ts

```diff
@@ -0,0 +1,264 @@
+import { Address, parseUnits } from 'viem';
+import { describe, expect, it, vi } from 'vitest';
+import chainlinkOracleAbi from '../abis/chainlinkOracle';
+import comptrollerAbi from '../abis/comptroller';
+import { CORE_COMPTROLLER_ADDRESS, CORE_MARKETS, ECOSYSTEM_MARKETS, ECOSYSTEM_UNITROLLER_ADDRESS, LIQUIDITY_DECIMALS, MARKET_DECIMALS, MarketProps } from '../constants';
+import { getMarketBorrowLimit } from './getMarketBorrowLimit';
+
+vi.mock('@heyanon/sdk');
+
+describe('getMarketBorrowLimit', () => {
+    describe.each`
+        marketType     | marketName | marketAddress
+        ${'core'}      | ${'USDC'}  | ${CORE_MARKETS.USDC}
+        ${'ecosystem'} | ${'USDC'}  | ${ECOSYSTEM_MARKETS.USDC}
+    `('marketType $marketType', ({ marketType, marketName, marketAddress }: MarketProps & { marketAddress: Address }) => {
+        it('should fail to fetch token price', async () => {
+            const props: Parameters<typeof getMarketBorrowLimit>[0] = {
+                account: '0x0000000000000000000000000000000000000000',
+                chainName: 'Avalanche',
+                ...({
+                    marketType,
+                    marketName,
+                } as MarketProps),
+            };
+
+            const oracleAddress = '0x0000000000000000000000000000000000000000';
```

Comment:
This time you are combining two different addresses into `0x0`.
I already mentioned the issue with zero address, but here in tracking balances you can mistakenly merge storage of sender and oracle contract.

Best practice would be still to avoid using `0x0` and avoid using same address for user and other contracts in tests 😉

Reply:
https://github.com/RealWagmi/anon-integration-guide/pull/16#discussion_r1935207820